gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -0,0 +1,215 @@
+      CopyableFile fileEntity =
+          builder.fileSet(fileSet).datasetOutputPath(this.fs.getUri().getPath()).build();
+      fileEntity.setSourceData(getSourceDataset());
+      fileEntity.setDestinationData(getDestinationDataset());
+      copyEntities.add(fileEntity);
+    }
+    log.info("{}.{} - generated {} copy entities", dbName, inputTableName, copyEntities.size());
+    return copyEntities;
+  }
+
+  /**
+   * Get builders for a {@link CopyableFile} for each file path
+   */
+  protected List<CopyableFile.Builder> getCopyableFilesFromPaths(Map<Path, FileStatus> pathToFileStatus, CopyConfiguration configuration) throws IOException {
+
+    List<CopyableFile.Builder> builders = Lists.newArrayList();
+    List<SourceAndDestination> dataFiles = Lists.newArrayList();
+    Configuration defaultHadoopConfiguration = new Configuration();
+    FileSystem actualSourceFs;
+
+    for (Map.Entry<Path, FileStatus> entry : pathToFileStatus.entrySet()) {
+      dataFiles.add(new SourceAndDestination(entry.getValue(), this.fs.makeQualified(entry.getKey())));
+    }
+
+    for (SourceAndDestination sourceAndDestination : dataFiles) {
+      actualSourceFs = sourceAndDestination.getSource().getPath().getFileSystem(defaultHadoopConfiguration);
+
+      // TODO: Add ancestor owner and permissions in future releases
+      builders.add(CopyableFile.fromOriginAndDestination(actualSourceFs, sourceAndDestination.getSource(),
+          sourceAndDestination.getDestination(), configuration));
+    }
+    return builders;
+  }
+  /**
+   * Finds all files of the Iceberg's current snapshot
+   * Returns a map of path, file status for each file that needs to be copied
+   */
+  protected Map<Path, FileStatus> getFilePathsToFileStatus() throws IOException {
+    Map<Path, FileStatus> result = Maps.newHashMap();
+    IcebergTable icebergTable = this.getIcebergTable();
+    IcebergSnapshotInfo icebergSnapshotInfo = icebergTable.getCurrentSnapshotInfo();
+
+    log.info("{}.{} - loaded snapshot '{}' from metadata path: '{}'", dbName, inputTableName,
+        icebergSnapshotInfo.getSnapshotId(), icebergSnapshotInfo.getMetadataPath());
+    List<String> pathsToCopy = icebergSnapshotInfo.getAllPaths();
+
+    for (String pathString : pathsToCopy) {
+      Path path = new Path(pathString);
+      result.put(path, this.fs.getFileStatus(path));
+    }
+    return result;
+  }
+
+  DatasetDescriptor getSourceDataset() {
+    return getDatasetDescriptor(sourceMetastoreURI);
+  }
+
+  DatasetDescriptor getDestinationDataset() {
+    return getDatasetDescriptor(targetMetastoreURI);
+  }
+
+  @NotNull
+  private DatasetDescriptor getDatasetDescriptor(Optional<String> stringMetastoreURI) {
+    String destinationTable = this.getDbName() + "." + this.getInputTableName();
+
+    URI hiveMetastoreURI = null;
+    if (stringMetastoreURI.isPresent()) {
+      hiveMetastoreURI = URI.create(stringMetastoreURI.get());
+    }
```

Comment:
Small nit: this can be simplified as a ternary operation

Reply:
sure... converted it into a ternary operation and pushed the change